File: slp/backend-dotnet/Features/Dashboard/DashboardService.cs
using backend_dotnet.Features.Quiz;
using backend_dotnet.Features.User;

namespace backend_dotnet.Features.Dashboard;

public class DashboardService : IDashboardService
{
    private readonly IQuizRepository _quizRepository;
    private readonly IUserRepository _userRepository;
    private readonly IWordOfTheDayProvider _wordProvider;

    public DashboardService(
        IQuizRepository quizRepository,
        IUserRepository userRepository,
        IWordOfTheDayProvider wordProvider)
    {
        _quizRepository = quizRepository;
        _userRepository = userRepository;
        _wordProvider = wordProvider;
    }

    public async Task<WordOfTheDayDto> GetWordOfTheDayAsync()
    {
        // Delegate to provider (static list or DB)
        return await _wordProvider.GetWordOfTheDayAsync();
    }

    public async Task<List<TopQuizDto>> GetTopQuizzesAsync(int limit)
    {
        return await _quizRepository.GetTopQuizzesByAttemptsAsync(limit);
    }

    public async Task<UserStatsDto> GetUserStatsAsync(int userId)
    {
        return await _userRepository.GetUserStatsAsync(userId);
    }
}

File: slp/backend-java/src/main/java/com/app/backendjava/features/dashboard/DashboardService.java
package com.app.backendjava.features.dashboard;

import com.app.backendjava.features.quiz.QuizRepository; // Assumed existing
import com.app.backendjava.features.user.UserRepository; // Assumed existing
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
@RequiredArgsConstructor
public class DashboardService implements IDashboardService {

    private final QuizRepository quizRepository;
    private final UserRepository userRepository;
    private final IWordOfTheDayProvider wordProvider;

    @Override
    public WordOfTheDayDto getWordOfTheDay() {
        return wordProvider.getWordOfTheDay();
    }

    @Override
    public List<TopQuizDto> getTopQuizzes(int limit) {
        return quizRepository.getTopQuizzesByAttemptsAsync(limit);
    }

    @Override
    public UserStatsDto getUserStats(int userId) {
        return userRepository.getUserStatsAsync(userId);
    }
}